metabase/metabase
# Properly apply feature flags to advanced dashboard subscription filters

### Description

Properly applies and tests `advanced_config` feature flag to dashboard subscription filters
Without token | With token
--- | ---
![Screen Shot 2023-07-05 at 9 21 44 AM](https://github.com/metabase/metabase/assets/30528226/ce363053-bd62-4320-9070-e1397d182634) | ![Screen Shot 2023-07-05 at 9 22 01 AM](https://github.com/metabase/metabase/assets/30528226/8db9babc-6253-403d-b679-93b28193d84b)
![withouttoken](https://github.com/metabase/metabase/assets/30528226/36180ea7-be00-4e18-8670-e4188ec8191e) | ![withtoken](https://github.com/metabase/metabase/assets/30528226/c16a23e4-b172-4760-81f7-c4297dae32ab)

Also adds test coverage for the sharingSidebar and subcomponents which were pretty sparsely tested before

### How to verify

Setup
- run metabase ee without a token loaded
- set up at least 1 method of notification: email or slack (email is easier) in admin settings
- create a dashboard with at least one filter
- open email notification settings
- see that there is no option to enable a filter for notifications
- add a token with `advanced_config` token permissions
- go back to dashboard subscriptions for you dashboard and see that you can now add a filter for the subscription


### Checklist
